core/trino-main/src/main/java/io/trino/sql/planner/TranslationMap.java

```diff
@@ -214,9 +213,14 @@ public Expression rewriteIdentifier(Identifier node, Void context, ExpressionTre
             public Expression rewriteFunctionCall(FunctionCall node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
             {
                 if (analysis.isPatternRecognitionFunction(node)) {
-                    List<Expression> rewrittenArguments = node.getArguments().stream()
-                            .map(argument -> treeRewriter.rewrite(argument, null))
-                            .collect(toImmutableList());
+                    ImmutableList.Builder<Expression> rewrittenArguments = ImmutableList.builder();
+                    if (!node.getArguments().isEmpty()) {
+                        rewrittenArguments.add(treeRewriter.rewrite(node.getArguments().get(0), null));
+                        if (node.getArguments().size() > 1) {
+                            // do not rewrite the offset literal
+                            rewrittenArguments.add(node.getArguments().get(1));
+                        }
+                    }
```

Comment:
Is there a test that reproduces this issue?

Reply:
Yes, there is a test in `TestRowPatternMatchingInWindow.testNavigationPastFrameBounds()`.
The example is:
```
query = "SELECT measure OVER w " +
                "          FROM (VALUES " +
                "                   (1, 10), " +
                "                   (2, 20), " +
                "                   (3, 30), " +
                "                   (4, 30), " +
                "                   (5, 40)" +
                "               ) t(id, value) " +
                "                 WINDOW w AS ( " +
                "                   ORDER BY id " +
                "                   MEASURES %s AS measure " +
                "                   ROWS BETWEEN CURRENT ROW AND 1 FOLLOWING " +
                "                   PATTERN (A B) " +
                "                   DEFINE A AS A.value = NEXT(A.value) " +
                "                )";

        // row 5 - out of frame bounds
        assertThat(assertions.query(format(query, "NEXT(B.value, 1)")))
                .matches("VALUES " +
                        "     (CAST(null AS integer)), " +
                        "     (null), " +
                        "     (null), " +
                        "     (null), " +
                        "     (null) ");
```